konduit-serving-pipeline/src/test/java/ai/konduit/serving/pipeline/impl/step/TestRegressionOutputStep.java

```diff
@@ -0,0 +1,179 @@
+/*
+ *  ******************************************************************************
+ *  * Copyright (c) 2020 Konduit K.K.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Apache License, Version 2.0 which is available at
+ *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+package ai.konduit.serving.pipeline.impl.step;
+
+import ai.konduit.serving.pipeline.api.data.Data;
+import ai.konduit.serving.pipeline.api.data.NDArray;
+import ai.konduit.serving.pipeline.api.pipeline.Pipeline;
+import ai.konduit.serving.pipeline.impl.pipeline.SequencePipeline;
+import ai.konduit.serving.pipeline.impl.step.ml.ssd.regression.RegressionOutputStep;
+import org.junit.Test;
+
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class TestRegressionOutputStep {
+
+
+    @Test
+    public void testCaseNonBatchDouble() {
+        Map<String, Integer> hashMap = new HashMap<String, Integer>();
+        hashMap.put("a",0);
+        hashMap.put("c",2);
+        double[][] values = new double[][]{{0.1, 0.2, 0.3}};
+
+
+        Pipeline p = SequencePipeline.builder()
+                .add(new RegressionOutputStep()
+                        .inputName("preds")
+                        .names(hashMap))
+
+                .build();
+
+        // double values input
+        NDArray preds = NDArray.create(values);
+        Data in = Data.singleton("preds", preds);
+        Data out = p.executor().exec(in);
+        assertTrue((double)out.get("a") == 0.1);
+        assertTrue((double)out.get("c") == 0.3);
```

Comment:
use `assertEquals(0.1, out.getDouble("a"), 0.0);`
assertEquals is preferable whenever possible in tests due to better output and easier debugging.
For example, if this fails on CI we get nothing is logging. Whereas assertEquals will give us an error message to tell us that expected 0.1, got (for example) 0.2, which can help us debug quicker (especially if problems aren't reproducible locally for us).